drivers/display/display_nrf_led_matrix.c

```diff
@@ -0,0 +1,448 @@
+#if USE_PWM
+	NRF_PWM_Type *pwm;
+#endif
+	uint8_t rows[ROW_COUNT];
+	uint8_t cols[COL_COUNT];
+	uint8_t pixel_mapping[PIXEL_COUNT];
+};
+
+struct display_drv_data {
+#if USE_PWM
+	uint16_t seq;
+#else
+	uint8_t gpiote_ch;
+#endif
+	uint8_t pixel_idx;
+	uint8_t framebuf[PIXEL_COUNT];
+	uint8_t brightness;
+	bool    blanking;
+};
+
+static void set_pin(uint8_t pin_info, bool active)
+{
+	uint32_t value = active ? 1 : 0;
+
+	if (pin_info & ACTIVE_LOW_MASK) {
+		value = !value;
+	}
+	nrf_gpio_pin_write(pin_info & PSEL_MASK, value);
+}
+
+static int api_blanking_on(const struct device *dev)
+{
+	struct display_drv_data *dev_data = dev->data;
+	const struct display_drv_config *dev_config = dev->config;
+
+	if (!dev_data->blanking) {
+		nrf_timer_task_trigger(dev_config->timer, NRF_TIMER_TASK_STOP);
+		for (uint8_t i = 0; i < ROW_COUNT; ++i) {
+			set_pin(dev_config->rows[i], false);
+		}
+		for (uint8_t i = 0; i < COL_COUNT; ++i) {
+			set_pin(dev_config->cols[i], false);
+		}
+
+		dev_data->blanking = true;
+	}
+
+	return 0;
+}
+
+static int api_blanking_off(const struct device *dev)
+{
+	struct display_drv_data *dev_data = dev->data;
+	const struct display_drv_config *dev_config = dev->config;
+
+	if (dev_data->blanking) {
+		dev_data->pixel_idx = PIXEL_COUNT - 1;
+
+		nrf_timer_task_trigger(dev_config->timer, NRF_TIMER_TASK_CLEAR);
+		nrf_timer_task_trigger(dev_config->timer, NRF_TIMER_TASK_START);
+
+		dev_data->blanking = false;
+	}
+
+	return 0;
+}
+
+static void *api_get_framebuffer(const struct device *dev)
+{
+	struct display_drv_data *dev_data = dev->data;
+
+	return dev_data->framebuf;
+}
+
+static int api_set_brightness(const struct device *dev,
+			      const uint8_t brightness)
+{
+	struct display_drv_data *dev_data = dev->data;
+	uint8_t new_brightness = CLAMP(brightness, 1, BRIGHTNESS_MAX);
+	int16_t delta = (int16_t)new_brightness - dev_data->brightness;
+
+	dev_data->brightness = new_brightness;
+
+	for (uint8_t i = 0; i < PIXEL_COUNT; ++i) {
+		uint8_t old_val = dev_data->framebuf[i];
+
+		if (old_val) {
+			int16_t new_val = old_val + delta;
+
+			dev_data->framebuf[i] =
+				(uint8_t)CLAMP(new_val, 1, BRIGHTNESS_MAX);
+		}
+	}
+
+	return 0;
+}
+
+static int api_set_contrast(const struct device *dev,
+			    const uint8_t contrast)
+{
+	return -ENOTSUP;
+}
+
+static int api_set_pixel_format(const struct device *dev,
+				const enum display_pixel_format format)
+{
+	switch (format) {
+	case PIXEL_FORMAT_MONO01:
+		return 0;
+	default:
+		return -ENOTSUP;
+	}
+}
+
+static int api_set_orientation(const struct device *dev,
+			       const enum display_orientation orientation)
+{
+	switch (orientation) {
+	case DISPLAY_ORIENTATION_NORMAL:
+		return 0;
+	default:
+		return -ENOTSUP;
+	}
+}
+
+static void api_get_capabilities(const struct device *dev,
+				 struct display_capabilities *caps)
+{
+	caps->x_resolution = X_PIXELS;
+	caps->y_resolution = Y_PIXELS;
+	caps->supported_pixel_formats = PIXEL_FORMAT_MONO10;
```

Comment:
PIXEL_FORMAT_MONO01, at least it should match one in api_set_pixel_format(), L:162